src/tasks/shop-admin/Customers/BulkEditCustomers.ts

```diff
@@ -0,0 +1,76 @@
+                const modalTitle = await AdminCustomerListing.getCustomerBulkEditModalTitle(customerCount);
+                await ShopAdmin.expects(modalTitle).toBeVisible();
+                await ShopAdmin.expects(AdminCustomerListing.modalHeaderCheckbox).toBeChecked();
+
+                // Start bulk edit
+                await AdminCustomerListing.startBulkEditButton.click();
+                await AdminCustomerListing.bulkEditModal.waitFor({ state: 'hidden' });
+                const bulkEditPageHeadline = await AdminCustomerBulkEdit.getPageHeadline(customerCount);
+                await ShopAdmin.expects(bulkEditPageHeadline).toBeVisible();
+                if (accountData) {
+                    if (accountData.customerGroup) {
+                        await AdminCustomerBulkEdit.changeCustomerGroupCheckbox.click();
+                        await AdminCustomerBulkEdit.customerGroupInput.click();
+                        await AdminCustomerBulkEdit.page.getByText(accountData.customerGroup).click();
+                    }
+                    if (accountData.accountStatus !== undefined) {
+                        await AdminCustomerBulkEdit.changeAccountStatusCheckbox.click();
+                        if (accountData.accountStatus) {
+                            await AdminCustomerBulkEdit.accountStatusInput.click();
+                        }
+                    }
+                    if (accountData.language) {
+                        await AdminCustomerBulkEdit.changeLanguageCheckbox.click();
+                        await AdminCustomerBulkEdit.changeLanguageInput.click();
+                        await AdminCustomerBulkEdit.page.getByText(accountData.language).click();
+                    }
+                    if (accountData.replyToCustomerGroupRequest) {
+                        await AdminCustomerBulkEdit.replyToCustomerGroupRequest.click();
+                        await AdminCustomerBulkEdit.replyToCustomerGroupRequestInput.click();
+                        await AdminCustomerBulkEdit.filtersResultPopoverItemList.getByText(accountData.replyToCustomerGroupRequest).click();
+                    }
+                }
+                if (tagData) {
+                    await AdminCustomerBulkEdit.changeTagsCheckbox.click();
+                    await AdminCustomerBulkEdit.changeTypeSelect.click();
+                    await AdminCustomerBulkEdit.page.getByText(tagData.changeType).click();
+                    for (const tag of tagData.tags) {
+                        await AdminCustomerBulkEdit.enterTagsSelect.click();
+                        await AdminCustomerBulkEdit.page.getByText(tag).click();
+                    }
+                }
+                if (customFieldData) {
+                    const customFieldSet = await AdminCustomerBulkEdit.getCustomFieldLinkByName(customFieldData.customFieldSetName);
+                    while (!await customFieldSet.isVisible()) {
```

Comment:
Why you chose that approche and not directly find the customSetName by Name and click it?

Reply:
There can be some leftover customsets from other tests that's hiding mine, so it's kinda a precautionary step.

Reply:
mhh I my case I could manage that directly with the filter function:

  ```
await ShopAdmin.expects(customFieldSetTabContent.customFieldSetTab).toBeVisible();
  await ShopAdmin.expects(customFieldSetTabContent.customFieldSetTab).toHaveText(customFieldSetName);
  await customFieldSetTabContent.customFieldSetTab.click();
```

But you solution seems to be OK too...so up to you. You can resolve the thread if you like.